src/common/TracepointProvider.cc

```diff
@@ -36,8 +39,8 @@ void TracepointProvider::verify_config(const struct md_config_t *conf) {
     return;
   }
 
-  void *handle = dlopen(m_library.c_str(), RTLD_NOW);
-  if (handle != NULL) {
+  m_handle = dlopen(m_library.c_str(), RTLD_NOW);
+  if (m_handle != NULL) {
```

Comment:
Nit: I would now eliminate the ```m_enabled``` member variable and just use ```m_handle != nullptr``` to determine if it's enabled.